code/modules/vehicles/spacepod/_pod.dm

```diff
@@ -0,0 +1,280 @@
+		return
+	mob_enter(dropped, flags = NONE) // force occupancy
+	dropped.visible_message(span_warning("[dropped] is forced into [src] by [dropper]!"))
+
+/obj/vehicle/sealed/space_pod/welder_act(mob/living/user, obj/item/welder)
+	if(user.combat_mode || DOING_INTERACTION(user, src))
+		return
+	. = NONE
+	if(atom_integrity >= max_integrity)
+		balloon_alert(user, "no damage!")
+		return
+	if(!welder.tool_start_check(user, amount=1))
+		return
+	user.balloon_alert_to_viewers("repairing pod!")
+	audible_message(span_hear("You hear welding."))
+	while(atom_integrity < max_integrity) //19-20 seconds to repair an iron pod from almost 0 to full
+		if(welder.use_tool(src, user, 1 SECONDS, volume=50))
+			atom_integrity += min(/obj/vehicle/sealed/space_pod::max_integrity / 20, (max_integrity - atom_integrity))
+			audible_message(span_hear("You hear welding."))
+		else
+			break
+
+/obj/vehicle/sealed/space_pod/vv_get_dropdown()
+	. = ..()
+	VV_DROPDOWN_OPTION("", "---------")
+	VV_DROPDOWN_OPTION(VV_HK_PODPANEL, "Pod Debug Panel")
+
+/obj/vehicle/sealed/space_pod/vv_do_topic(list/href_list)
+	. = ..()
+	if(!.)
+		return
+
+	if(href_list[VV_HK_PODPANEL])
+		if(!check_rights(R_ADMIN))
+			return
+		SSadmin_verbs.dynamic_invoke_verb(usr, /datum/admin_verb/pod_debug_panel, src)
+
+/obj/vehicle/sealed/space_pod/proc/process_huds()
+	for(var/mob/living/occupant as anything in occupants)
+		if(cell?.maxcharge)
+			var/cellcharge = cell.charge/cell.maxcharge
+			switch(cellcharge)
+				if(0.75 to INFINITY)
+					occupant.clear_alert(ALERT_CHARGE)
+				if(0.5 to 0.75)
+					occupant.throw_alert(ALERT_CHARGE, /atom/movable/screen/alert/lowcell, 1)
+				if(0.25 to 0.5)
+					occupant.throw_alert(ALERT_CHARGE, /atom/movable/screen/alert/lowcell, 2)
+				if(0.01 to 0.25)
+					occupant.throw_alert(ALERT_CHARGE, /atom/movable/screen/alert/lowcell, 3)
+				else
+					occupant.throw_alert(ALERT_CHARGE, /atom/movable/screen/alert/emptycell)
+		else
+			occupant.throw_alert(ALERT_CHARGE, /atom/movable/screen/alert/nocell)
+		var/integrity = get_integrity_percentage() * 100
+		switch(integrity)
+			if(40 to 60)
+				occupant.throw_alert(ALERT_MECH_DAMAGE, /atom/movable/screen/alert/pod_damage, 1)
+			if(20 to 40)
+				occupant.throw_alert(ALERT_MECH_DAMAGE, /atom/movable/screen/alert/pod_damage, 2)
+			if(-INFINITY to 20)
+				occupant.throw_alert(ALERT_MECH_DAMAGE, /atom/movable/screen/alert/pod_damage, 3)
+			else
+				occupant.clear_alert(ALERT_MECH_DAMAGE)
+
+// handles giving equipment actions + some other stuff
+/obj/vehicle/sealed/space_pod/after_add_occupant(mob/living/occupant)
+	. = ..()
+	occupant.refresh_gravity()
+	if(length(occupants) == 1) //first occupant only
+		panel_open = FALSE //automatic screws,,,, waow....
+		cycle_tank_air()
+	for(var/obj/item/pod_equipment/equipment as anything in get_all_parts())
+		var/datum/action/action = equipment.create_occupant_actions(occupant, occupants[occupant])
+		if(isnull(action))
+			continue
+		if(islist(action))
```

Comment:
I hate vehicle code

Reply:
yeah me too but i dont know if this is an actual review